src/monodroid/jni/monodroid-glue.c

```diff
@@ -3031,13 +3031,13 @@ _monodroid_get_display_dpi (float *x_dpi, float *y_dpi)
 }
 
 static void
-lookup_bridge_info (MonoImage *image, const MonoJavaGCBridgeType *type, MonoJavaGCBridgeInfo *info)
+lookup_bridge_info (MonoDomain *domain, MonoImage *image, const MonoJavaGCBridgeType *type, MonoJavaGCBridgeInfo *info)
 {
-	info->klass             = mono.mono_class_from_name (image, type->namespace,    type->typename);
-	info->handle            = mono.mono_class_get_field_from_name (info->klass,     "handle");
-	info->handle_type       = mono.mono_class_get_field_from_name (info->klass,     "handle_type");
-	info->refs_added        = mono.mono_class_get_field_from_name (info->klass,     "refs_added");
-	info->weak_handle       = mono.mono_class_get_field_from_name (info->klass,     "weak_handle");
+	info->klass             = monodroid_get_class_from_image (&mono, domain, image, type->namespace, type->typename);
```

Comment:
From what I read through the code `MonoImage` and `MonoClass` are shared among domain instances thus they are not specific to one.

What can happen though is that if all domains that reference an image are unloaded then that image and its class instances become invalid.

@vargaz can probably give us more info on this.